src/coreclr/gc/gcpriv.h

```diff
@@ -5105,23 +5105,24 @@ size_t& generation_allocated_since_last_pin (generation* inst)
 #endif //FREE_USAGE_STATS
 
 inline
-float generation_allocator_efficiency (generation* inst)
+size_t generation_allocator_efficiency_percent (generation* inst)
 {
-    // Because of speculative execution, it is not always safe to do the following code if sum is equal to zero
-    // if (sum!=0.0f)
-    //   return ... / sum;
-    // To prevent this, add a small value to the divider.
-    // It will not change the result if sum is not zero because in this
-    // case sum>1.0 and precision of float is limited to 7 digits
-
-    float sum = (float)(generation_free_list_allocated (inst) + generation_free_obj_space (inst));
-    return ((float) (generation_free_list_allocated (inst))) / (sum + 1.0e-10f);
+    size_t free_obj_space = generation_free_obj_space (inst);
+    size_t free_list_allocated = generation_free_list_allocated (inst);
+    if (free_list_allocated==0)
+      return 0;
+    return (100 * free_list_allocated) / (free_list_allocated + free_obj_space);
```

Comment:
Can this hit integer overflow?

Reply:
if `size_t` is 64bit, it can not overflow because `free_list_allocated` will never be greater than `2^60`.
I can use `uint64_t` if you prefer.

Reply:
@grospelliergilles, do you plan to update the PR with `uint64_t` and perhaps we can evaluate again.

Reply:
Thanks for you reply @mangod9. Yes I will update my PR if you want. I will do that this week